### DISCUSSION TITLE
How to setting the TurbSim inp file(Usabletime, Analysistime)

### DISCUSSION THREAD

**Discussion by onibini** [OpenFAST/openfast#2542] [Q&A] [answered]
I'm having trouble editing the TurbSim input file, so I'm asking a question.

![KakaoTalk_20241201_212832389](https://github.com/user-attachments/assets/892d880e-ce66-4e2d-b380-84383935b096)

I was setting like this picture.

So what I'm wondering is this
If I run openfast as 1000 seconds, 
I don't know how to set usabletime and analysistime.
Of course, I checked the manual, but it's hard to understand,
so I would appreciate it if you could tell me how to change the settings when running 1000 seconds.

**Comment by jjonkman** [comment, marked as answer]
Dear @onibini,

In TurbSim, `AnalysisTime` is the length of time turbulent wind is generated internally and `UsableTime` is the length of time written to the TurbSim output file.  You can't have `AnalysisTime` be less than `UsableTime` (as you are showing).

For normal use where OpenFAST will be run for 600-s (10-min) after the start-up transient, I would recommend setting `AnalysisTime` = 600 s and `UsableTime` = "ALL", which will ensure TurbSim outputs all of its data and the InflowWind module of OpenFAST will interpret the data as periodic, which will allow you to run OpenFAST simulations of 600 s + some start-up transient time (typically 30-100 s, so, OpenFAST is run for `TMax` = 630-700 s with `TStart` = 30-100 s to throw out that start-up transient data).  If you are aiming for 1000 s of actual OpenFAST output, change this prescription to `AnalysisTime` = 1000 s and `TMax` = 1030-1100 s.

Best regards,

**Reply by onibini** [reply]
Thank you very much for your reply. It helped me a lot in my research.
